pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/oxia/OxiaMetadataStore.java

```diff
@@ -0,0 +1,283 @@
+                        children ->
+                                children.stream().map(child -> child.substring(pathWithSlash.length())).toList())
+                .exceptionallyCompose(this::convertException);
+    }
+
+    @Override
+    protected CompletableFuture<Boolean> existsFromStore(String path) {
+        return client.get(path).thenApply(Objects::nonNull)
+                .exceptionallyCompose(this::convertException);
+    }
+
+    @Override
+    protected CompletableFuture<Optional<GetResult>> storeGet(String path) {
+        return client.get(path).thenApply(res -> convertGetResult(path, res))
+                .exceptionallyCompose(this::convertException);
+    }
+
+    @Override
+    protected CompletableFuture<Void> storeDelete(String path, Optional<Long> expectedVersion) {
+        return getChildrenFromStore(path)
+                .thenCompose(
+                        children -> {
+                            if (children.size() > 0) {
+                                return CompletableFuture.failedFuture(
+                                        new MetadataStoreException("Key '" + path + "' has children"));
+                            } else {
+                                var delOption =
+                                        expectedVersion
+                                                .map(DeleteOption::ifVersionIdEquals)
+                                                .orElse(DeleteOption.Unconditionally);
+                                CompletableFuture<Boolean> result = client.delete(path, delOption);
+                                return result
+                                        .thenCompose(
+                                                exists -> {
+                                                    if (!exists) {
+                                                        return CompletableFuture.failedFuture(
+                                                                new MetadataStoreException.NotFoundException(
+                                                                        "Key '" + path + "' does not exist"));
+                                                    }
+                                                    return CompletableFuture.completedFuture((Void) null);
+                                                })
+                                        .exceptionallyCompose(this::convertException);
+                            }
+                        });
+    }
+
+    @Override
+    protected CompletableFuture<Stat> storePut(
+            String path, byte[] data, Optional<Long> optExpectedVersion, EnumSet<CreateOption> options) {
+        CompletableFuture<Void> parentsCreated = createParents(path);
+        return parentsCreated.thenCompose(
+                __ -> {
+                    var expectedVersion = optExpectedVersion;
+                    if (expectedVersion.isPresent()
+                            && expectedVersion.get() != -1L
```

Comment:
nit: is it better to have a constant for the magic value -1 ?

Reply:
This is used across all plugins. The metadata store interface takes -1 as no version, following the ZK api. We should change for all plugins.